### PR TITLE
Entity as another entity's field

### LINKED ISSUE #43: Error using a entity as a field of another entity
**Describe the bug**
When I use a entity as a field of another entity, for some reason it threw an Unknow type error.

**To Reproduce**
Create an entity 
Import this entity into another entity
Use this imported entity as a field

example:
```
const { field, entity } = require('@herbsjs/herbs')
const B = require('../entities')

const A = entity("A", {
    B: field(B)
})

module.exports = A
```

Update the entities index

example:
```
module.exports = {
    A: require('./A'),
    B: require('./B')
}
```

Check the log when you run the application

**Expected behavior**
Graphql should create type and input without error

**Screenshots**
imagem erro

### PR DESCRIPTION
**Pull request by pamellaas**
Fixes #43

It adds support to use an entity as a field of another entity without error

## Proposed Changes

1. Added a parameter in the functions `entityField2gql` and `entityFieldType2gql`, so that it can identify when the function is creating a type and when it is an input type

### Author/Contributor
- [x] If documentation is needed for this change, has that been included in this pull request
- [x] Remember to check if code coverage decrease, if so, please implement the necessary tests

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by jhomarolo** [comment]
@pamellaas merge master into your branch and then delete your package.lock file.
After deleting the file, run `npm install`, add the changer files and make a new push, please.

File: test/entity2input.test.js [hunk 1]
```diff
@@ -46,6 +46,35 @@ dateArrayField: [Date]
       )
     })
 
+     it("should convert an entity to input when entity has entity references", async () => {
```
Comment (dalssoft) [inline test/entity2input.test.js:49]: this test checks outputs for 'entity2input'. there should be the same test for 'entity2type'

**Review by dalssoft — CHANGES_REQUESTED** [review]

**Comment by dalssoft** [comment]
I don´t get it. This PR is suppose to handle entities as type for other entities. However, from what I could read from the code it is a implementation of 'Input' types. How does this fix the problem?

**Comment by Vx-Nico** [comment]
> I don´t get it. This PR is suppose to handle entities as type for other entities. However, from what I could read from the code it is a implementation of 'Input' types. How does this fix the problem?

In the issue we imported the "B" entity in the wrong way, we should have done this way : 
`const { B } = require('../entities')`

When we import in that way, it trow this error : 
image (9)

This is the problem that we solved, by adding a new parameter when calling the function **entityField2gql** and creating a new if statement in the **entityFieldType2gql** function.

File: test/entity2input.test.js (hunk 1 above)
Comment (Vx-Nico) [reply test/entity2input.test.js:49]: We believe that already exists, here is the link : https://github.com/herbsjs/herbs2gql/blob/master/test/entity2type.test.js#L126

**Review by dalssoft — APPROVED** [review]

**Comment by jhomarolo** [comment]
@pamellaas, @Vx-Nico it seems this version is not working properly.

Could you please fix this asap? https://github.com/herbsjs/todolist-on-herbs/issues/106

**Pull request merged**